pkg/controller/jobframework/reconciler.go

```diff
@@ -273,11 +273,8 @@ func (r *JobReconciler) ReconcileGenericJob(ctx context.Context, req ctrl.Reques
 	}
 
 	if wl != nil && apimeta.IsStatusConditionTrue(wl.Status.Conditions, kueue.WorkloadFinished) {
-		// Finalize the job if it's finished
-		if _, finished := job.Finished(); finished {
-			if err := r.finalizeJob(ctx, job); err != nil {
-				return ctrl.Result{}, err
-			}
+		if err := r.finalizeJob(ctx, job); err != nil {
+			return ctrl.Result{}, err
 		}
```

Comment:
Since this PR, we always finalize the Job when the Job has WorkloadFinished condition.
So, is this workload-controller mechanism still needed?

https://github.com/kubernetes-sigs/kueue/blob/5023dd227e881936d4b2733d91a30c169426053b/pkg/controller/core/workload_controller.go#L158-L160

Reply:
It still helps in the case of a rogue user manually deleting finalizers.

But I'm also never sure which cases we are missing when dealing with finalizers. So I prefer to keep it.

Reply:
That makes sense.
Thank you for the clarification!
/lgtm